src/python_testing/TC_OpstateCommon.py

```diff
@@ -0,0 +1,1191 @@
+            self.device = "Generic"
+        elif self.test_info.cluster == Clusters.OvenCavityOperationalState:
+            self.device = "Oven"
+        else:
+            asserts.fail("Cluster not supported!")
+
+    def init_test(self):
+        self.is_ci = self.check_pics("PICS_SDK_CI_ONLY")
+        if self.is_ci:
+            app_pid = self.matter_test_config.app_pid
+            if app_pid == 0:
+                app_pid = get_pid("chip-all-clusters-app")
+                if app_pid is None:
+                    asserts.fail("The --app-pid flag must be set when PICS_SDK_CI_ONLY is set")
+            self.app_pipe = self.app_pipe + str(app_pid)
+
+    # Sends and out-of-band command to test-app
+    def write_to_app_pipe(self, command):
+        with open(self.app_pipe, "w") as app_pipe:
+            app_pipe.write(command + "\n")
+
+    def send_manual_or_pipe_command(self, command):
+        if self.is_ci:
+            self.write_to_app_pipe(command)
+        else:
+            input("Press Enter when done.\n")
+
+    async def send_cmd(self, endpoint, cmd, timedRequestTimeoutMs=None):
+        logging.info("\t##### Command %s" % cmd)
+
+        try:
+            return await self.send_single_cmd(cmd=cmd,
+                                              endpoint=endpoint,
+                                              timedRequestTimeoutMs=timedRequestTimeoutMs)
+
+        except InteractionModelError as e:
+            asserts.fail("Unexpected error returned: %s" % e.status)
+
+    async def send_cmd_expect_response(self, endpoint, cmd, expected_response, timedRequestTimeoutMs=None):
+        ret = await self.send_cmd(endpoint=endpoint,
+                                  cmd=cmd,
+                                  timedRequestTimeoutMs=timedRequestTimeoutMs)
+
+        asserts.assert_equal(ret.commandResponseState.errorStateID,
+                             expected_response,
+                             "Current value not according expected. Expected %d, received %d!" %
+                             (expected_response, ret.commandResponseState.errorStateID))
+
+    async def read_expect_success(self, endpoint, attribute):
+        logging.info("\t##### Read %s" % attribute)
+        attr_value = await self.read_single_attribute_check_success(endpoint=endpoint,
+                                                                    cluster=self.test_info.cluster,
+                                                                    attribute=attribute)
+        logging.info("\t## %s: %s" % (attribute, attr_value))
+
+        return attr_value
+
+    async def read_and_expect_value(self, endpoint, attribute, expected_value):
+        attr_value = await self.read_expect_success(
+            endpoint=endpoint,
+            attribute=attribute)
+
+        asserts.assert_equal(attr_value, expected_value,
+                             "Current value not according expected. Expected %d, received %d!" %
```

Comment:
- Nit: Grammar on "Current value doesn't match expected".
- Consider actually logging the attribute and endpoint, which are very important to diagnose causes.
- The expected/actual will be shown to you by the `asserts` module, for all assertions: https://github.com/google/mobly/blob/master/mobly/asserts.py#L56
- Consider using formatted "f-strings" (which would look something like: `f"Value mismatched from expectation for {attribute} on {endpoint:d}"`.

This comment applies several different places.

Reply:
done